examples/transforms/global_vs_local_translation.rs

```diff
@@ -0,0 +1,140 @@
+use bevy::prelude::*;
+
+// Define a struct to store basic state information.
+struct EntityState {
+    spawn_location: Vec3,
+    stopped: bool,
+}
+
+// Define a marker for entities that should stop at a distance in respect to their global transform.
+struct GlobalStop;
+// Define a marker for entities that should stop at a distance in respect to their local transform.
+struct LocalStop;
+
+// Define the maximum distance an entity should be able to move away from its spawn.
+const MAX_DISTANCE: f32 = 5.0;
+
+// Ease up creation of EntityState structs.
+impl EntityState {
+    fn new(spawn_location: Vec3) -> Self {
+        EntityState {
+            spawn_location,
+            stopped: false,
+        }
+    }
+}
+
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(move_cubes.system())
+        .add_system(stop_too_far_global_distance.system())
+        .add_system(stop_too_far_local_distance.system())
+        .run();
+}
+
+// Startup system to setup the scene and spawn all relevant entities.
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // To show the difference between a local transform (rotation, scale and position in respect to a given entity)
+    // and global transform (rotation, scale and position in respect to the base coordinate system of the visible scene)
+    // it's helpful to add multiple entities that are attached to each other.
+    // This way we'll see that the transform in respect to an entities parent is different to the
+    // global transform within the visible scene.
+    // This example focuses on translation only to make a simple case for the differences.
+    let main_entity_spawn: Transform = Transform::from_translation(Vec3::ZERO);
+
+    // Spawn a basic cube to have an entity as reference.
+    let mut main_entity = commands.spawn_bundle(PbrBundle {
+        mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
+        material: materials.add(Color::YELLOW.into()),
+        transform: main_entity_spawn,
+        ..Default::default()
+    });
+    main_entity
+        .insert(GlobalStop)
+        .insert(EntityState::new(main_entity_spawn.translation));
```

Comment:
```suggestion
    let mut main_entity = commands.spawn_bundle(PbrBundle {
        mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
        material: materials.add(Color::YELLOW.into()),
        transform: main_entity_spawn,
        ..Default::default()
    })
        .insert(GlobalStop)
        .insert(EntityState::new(main_entity_spawn.translation));
```
This will need a run of `cargo fmt` afterwards.

Reply:
I initially thought so too but due to the usage for spawning children afterwards the compiler won't let us do this.
error[E0716]: temporary value dropped while borrowed
  --> examples/transforms/global_vs_local_translation.rs:52:23
[...]
79 |       main_entity.with_children(|child_builder| {
   |       ----------- borrow later used here
   |
   = note: consider using a `let` binding to create a longer lived value

Reply:
Oh right, in that case it is fine as is.
You could do, but i dont know if that is better:
```rust
    let mut main_entity = commands.spawn();
    main_entity
        .insert_bundle(PbrBundle {
            mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
            material: materials.add(Color::YELLOW.into()),
            transform: main_entity_spawn,
            ..Default::default()
        })
        .insert(GlobalStop)
        .insert(EntityState::new(main_entity_spawn.translation));
```

Reply:
I'd do it. keeps the structure similar to other examples.